src/postgres_client/postgres_client_account_index.rs

```diff
@@ -0,0 +1,377 @@
+use {
+    super::{
+        DbAccountInfo, ReadableAccountInfo, SimplePostgresClient,
+        DEFAULT_ACCOUNTS_INSERT_BATCH_SIZE,
+    },
+    crate::{
+        accountsdb_plugin_postgres::{
+            AccountsDbPluginPostgresConfig, AccountsDbPluginPostgresError,
+        },
+        inline_spl_token::{self, GenericTokenAccount},
+        inline_spl_token_2022,
+    },
+    log::*,
+    postgres::{Client, Statement},
+    solana_accountsdb_plugin_interface::accountsdb_plugin_interface::AccountsDbPluginError,
+    solana_measure::measure::Measure,
+    solana_metrics::*,
+    solana_sdk::pubkey::Pubkey,
+    tokio_postgres::types,
+};
+
+const TOKEN_INDEX_COLUMN_COUNT: usize = 3;
+/// Model the secondary index for token owner and mint
+pub struct TokenSecondaryIndex {
+    owner: Vec<u8>,
+    account_key: Vec<u8>,
+    slot: i64,
+}
+
+impl SimplePostgresClient {
+    pub fn build_single_token_owner_index_upsert_statement(
+        client: &mut Client,
+        config: &AccountsDbPluginPostgresConfig,
+    ) -> Result<Statement, AccountsDbPluginError> {
+        let stmt =
+            "INSERT INTO spl_token_owner_index AS owner_index (owner_key, account_key, slot) \
+        VALUES ($1, $2, $3) \
+        ON CONFLICT (owner_key, account_key) \
+        DO UPDATE SET slot=excluded.slot \
+        WHERE owner_index.slot < excluded.slot";
```

Comment:
can we make these const strings, i.e. const BULK_TOKEN_OWNER_INDEX_INSERT_STATEMENT: &str = "";

Reply:
Done